core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleParser.java

```diff
@@ -304,7 +305,7 @@ protected void parsePrefixID() throws IOException, RDFParseException, RDFHandler
 				throwEOFException();
 			}
 
-			prefixID.append(Character.toChars(c));
+			Chars.append(prefixID, c);
```

Comment:
I will submit a follow up patch to this to reuse your method for prefixID here. It should be okay as it is a distinct branch in parseDirective (but the call to prefixID.toString below would need to be above the call to parseURI that would otherwise clear its StringBuilder instance.